pkg/api/v1/types.go

```diff
@@ -2449,6 +2449,50 @@ type SeedSpec struct {
 	EtcdBackupRestore *kubermaticv1.EtcdBackupRestore `json:"etcdBackupRestore,omitempty"`
 }
 
+// CreateSeedSpec is the structure that is used to create seed.
+// swagger:model CreateSeedSpec
+type CreateSeedSpec struct {
+	// Optional: Country of the seed as ISO-3166 two-letter code, e.g. DE or UK.
+	// For informational purposes in the Kubermatic dashboard only.
+	Country string `json:"country,omitempty"`
+	// Optional: Detailed location of the cluster, like "Hamburg" or "Datacenter 7".
+	// For informational purposes in the Kubermatic dashboard only.
+	Location string `json:"location,omitempty"`
+	// The raw Kubeconfig encoded to base64. This field is used for cluster creation or update.
+	Kubeconfig string `json:"kubeconfig"`
+	// Optional: This can be used to override the DNS name used for this seed.
+	// By default the seed name is used.
+	SeedDNSOverwrite string `json:"seed_dns_overwrite,omitempty"`
+	// Optional: ProxySettings can be used to configure HTTP proxy settings on the
+	// worker nodes in user clusters. However, proxy settings on nodes take precedence.
+	ProxySettings *CreateSeedProxySettings `json:"proxy_settings,omitempty"`
+	// Optional: ExposeStrategy explicitly sets the expose strategy for this seed cluster, if not set, the default provided by the master is used.
+	ExposeStrategy string `json:"expose_strategy,omitempty"`
```

Comment:
why not make this `ExposteStrategy kubermaticv1.ExposeStrategy`, then no need for the switch in the Seed generation?